src/main/java/io/xdb/core/encoder/ObjectEncoder.java

```diff
@@ -12,21 +17,38 @@ public interface ObjectEncoder {
     String getEncodingType();
 
     /**
-     * Encode a Java object to an EncodedObject
+     * Encode a Java object to a string.
      *
-     * @param object Java object to convert
-     * @return encoded object with the encoding type of the encoder
+     * @param object Java object to convert.
+     * @return encoded string.
      */
-    EncodedObject encode(final Object object);
+    String encodeToString(final Object object);
```

Comment:
I would prefer to keep things separate. 
The general object encoder is for data serialization(user may choose json/grpc/binary etc) different protocol to searlizae an object. 

While for global attribute, it's for converting into a string value that can be directly executing as SQL(or mongo QL etc). 

While we support default implementation here, users will have customization for different purpose and they won't mix them together.